jerry-core/parser/js/js-parser-expr.c

```diff
@@ -480,9 +481,10 @@ parser_parse_class_literal (parser_context_t *context_p) /**< context */
       lexer_literal_t *literal_p = (lexer_literal_t *) parser_list_append (context_p, &context_p->literal_pool);
       literal_p->type = LEXER_UNUSED_LITERAL;
       literal_p->status_flags = 0;
+      uint16_t result_index = context_p->literal_count;
       literal_p->u.bytecode_p = parser_parse_function (context_p, constructor_status_flags);
       literal_p->type = LEXER_FUNCTION_LITERAL;
-      parser_emit_cbc_literal (context_p, PARSER_TO_EXT_OPCODE (CBC_EXT_SET_CLASS_LITERAL), context_p->literal_count);
+      parser_emit_cbc_literal (context_p, PARSER_TO_EXT_OPCODE (CBC_EXT_SET_CLASS_LITERAL), result_index);
```

Comment:
I don't think that this change is necessary. Where else is `result_index` used?

Reply:
It should be. That was currently an unrevealed bug, since the `parser_parse_function` can overwrite the literal_object and increase the literal count as well. But thanks for the reminding I'll add a test case for it.

Reply:
I think I see what I've missed here: is it `parser_parse_function` that may update `context_p->literal_count` and that's why it should be saved? If so, it's OK. But I think it would be better not to inject `result_index` in the middle of `literal_p->field` assignments. It could be simple moved two lines upwards.

Reply:
Exactly. Okay it can me moved up.